src/Autofac/Util/DeterministicGuid.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Linq;
+using System.Security.Cryptography;
+using System.Text;
+#pragma warning disable CA5350 // Do Not Use Weak Cryptographic Algorithms - SHA1 required by specification
+
+namespace Autofac.Util
+{
+    /// <summary>
+    /// Utility class for generating predictable <see cref="Guid"/>s, per RFC 4122 §4.3.
+    /// </summary>
+    internal static class DeterministicGuid
```

Comment:
This class is based on things I've done for my day job, which was derived from some stackoverflow posts. I wasn't sure what the practice was for comments regarding attribution/credit, but I can definitely add the SO answer for posterity.